Training/Model/AdminSession.php

```diff
@@ -0,0 +1,197 @@
+<?php
+/**
+ * Codifi_Training
+ *
+ * @copyright   Copyright (c) 2021 Codifi
+ * @author      Pavel Zelenevich <pzelenevich@codifi.me>
+ */
+
+declare(strict_types=1);
+
+namespace Codifi\Training\Model;
+
+use Codifi\Training\Model\ConfigProvider;
+use Magento\Backend\Model\Auth\Session;
+
+/**
+ * Class AdminSession
+ * @package Codifi\Training\Model
+ */
+class AdminSession
+{
+    /**
+     * Auth session
+     *
+     * @var Session
+     */
+    private $authSession;
+
+    /**
+     * Backend session
+     *
+     * @var \Magento\Backend\Model\Session
+     */
+    protected $backendSession;
+
+    /**
+     * Config provider
+     *
+     * @var ConfigProvider
+     */
+    private $configProvider;
+
+    /**
+     * AdminSession constructor.
+     *
+     * @param \Codifi\Training\Model\ConfigProvider $configProvider
+     * @param Session $authSession
+     * @param \Magento\Backend\Model\Session $backendSession
+     */
+    public function __construct(
+        ConfigProvider $configProvider,
+        Session $authSession,
+        \Magento\Backend\Model\Session $backendSession
+    ) {
+        $this->configProvider = $configProvider;
+        $this->authSession = $authSession;
+        $this->backendSession = $backendSession;
+    }
+
+    /**
+     * Get credit_hold attribute value from current customer.
+     *
+     * @return bool
+     */
+    public function getCustomerAttr() : bool
+    {
+        $customerData = $this->backendSession->getCustomerData();
+        $customerAttribute = $customerData['account'];
+        if ($customerAttribute !== null) {
+            $value = (bool)$customerAttribute[ConfigProvider::ATTRIBUTE_CODE_CREDIT_HOLD];
+        } else {
+            $value = 0;
+        }
+
+        return $value;
+    }
+
+    /**
+     * Get current customer id from admin session.
+     *
+     * @return int
+     */
+    public function getCustomerId() : int
+    {
+        $customerData = $this->backendSession->getCustomerData();
+        $customerAttribute = $customerData['account'];
+        if ($customerAttribute !== null) {
+            $value = $customerAttribute['id'];
+        } else {
+            $value = 0;
+        }
+
+        return $value;
+    }
+
+    /**
+     * Set customer id to array in admin session.
+     *
+     * @return void
+     */
+    public function setCustomerIdToAdminSession() : void
```

Comment:
setCustomerIdToAdminSession and setCustomerIdToAdminSessionFromSave can refactored to one method
For example:
```
public function setCustomerIdToAdminSession(int $customerId = null) 
{
    if (!$customerId) {
        $customerId = $this->getCustomerId();
    }
    
    $customerIds = $this->getCustomerIds() ?? [];
    $customerIds[] = $customerId;

    $this->authSession->setCustomerIds(array_unique($customerIds));
}
```